File: javascript/fibonacci.js
function fibonacci(num) {
  if (num < 2) {
    return num
  }
  let count = 2
  let first = 1
  let second = 2
  while (count <= num) {
    if (count ===  num) {
      return first
    }
    second += first
    first = second - first
    count++
  }
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("Expecting: 1");
  console.log("=>", fibonacci(1));

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("Expecting: 2");
  console.log("=>", fibonacci(3));
  
  console.log("Expecting: 3");
  console.log("=>", fibonacci(4));

  console.log("Expecting: 5");
  console.log("=>", fibonacci(5));

  console.log("Expecting: 8");
  console.log("=>", fibonacci(6));

  console.log("Expecting: 13");
  console.log("=>", fibonacci(7));

  console.log("Expecting: 21");
  console.log("=>", fibonacci(8));

  console.log("Expecting: 34");
  console.log("=>", fibonacci(9));

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  if num < 2 
    return num
  end
  count = 2
  f = 1
  s = 2
  while count <= num do
    if count == num 
      return f
    end
    s += f
    f = s - f
    count += 1
  end
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts "Expecting: 3"
  puts "=>", fibonacci(4)

  puts "Expecting: 5"
  puts "=>", fibonacci(5)

  puts "Expecting: 8"
  puts "=>", fibonacci(6)

  puts "Expecting: 13"
  puts "=>", fibonacci(7)

  puts "Expecting: 21"
  puts "=>", fibonacci(8)

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  puts "Expecting: 89"
  puts "=>", fibonacci(11)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
